File: Client/src/app/shared/models/customerBasket.ts
import { v4 as uuidv4 } from 'uuid'
import { ICustomerBasketItem } from "./CustomerBasketItem";

export interface ICustomerBasket {
  id: string;
  items: ICustomerBasketItem[]
}

export class CustomerBasket implements ICustomerBasket {
  id = uuidv4();
  items: ICustomerBasketItem[] = [];
}

File: Models/CustomerBasket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace KPI.SportStuffInternetShop.Models {
    public class CustomerBasket {
        [Required]
        public Guid Id { get; set; }

        public Guid? DeliveryMethodId { get; set; }

        public string ClientSecret { get; set; }

        public string PaymentIntentId { get; set; }

        public List<CustomerBasketItem> Items { get; set; }
    }
}
